torch/nn/functional.py

```diff
@@ -2511,7 +2511,37 @@ def smooth_l1_loss(input, target, size_average=None, reduce=None, reduction='mea
     if size_average is not None or reduce is not None:
         reduction = _Reduction.legacy_get_string(size_average, reduce)
     if target.requires_grad:
-        ret = _smooth_l1_loss(input, target)
+        ret = _huber_loss(input, target)
+        if reduction != 'none':
+            ret = torch.mean(ret) if reduction == 'mean' else torch.sum(ret)
+    else:
+        expanded_input, expanded_target = torch.broadcast_tensors(input, target)
+        ret = torch._C._nn.smooth_l1_loss(expanded_input, expanded_target, _Reduction.get_enum(reduction))
+    return ret
+
+
+def huber_loss(input, target, delta, size_average=None, reduce=None, reduction='mean'):
```

Comment:
This looks like the same as `smooth_l1_loss` but for the `delta` parameter. Let's just add a new parameter `delta` with default value `1`.